src/components/App.tsx

```diff
@@ -90,15 +86,21 @@ class App extends React.Component<RouteProps, State> {
         this.setState({
           isLoading: false,
           networkErrorOccurred: true,
-          errorMessage
+          errorMessage,
+          logEntries: []
         });
       });
-  }
 
-  private handleAddEntry = (entry: LiftLogEntry) =>
-    this.liftLogService
-      .addEntry(this.logName, entry)
-      .then(() => this.reloadLifts());
+  private async handleAddEntry(entry: LiftLogEntry) {
+    this.setState(prevState => ({
+      isLoading: true,
+      loadingMessage: `Adding entry for ${entry.name}...`,
+      logEntries: prevState.logEntries
```

Comment:
You don't have to explicitly transfer `logEntries` from `prevState` onto the new state, so you can just remove `prevState` and just call `setState` with an object.

Reply:
That doesn't pass the typechecker. Or should I spread the `prevState`?